docs/architecture/adr-028-public-key-addresses.md

```diff
@@ -100,69 +181,44 @@ message PubKey {
   bytes key = 1;
 }
 ```
- 
+
 All protobuf messages have unique fully qualified names, in this example `cosmos.crypto.sr25519.PubKey`.
 These names are derived directly from .proto files in a standardized way and used
 in other places such as the type URL in `Any`s. Since there is an easy and obvious
 way to get this name for every protobuf type, we can use this message name as the
 key type `prefix` when creating addresses. For all basic public keys, `contents`
 should just be the raw unencoded public key bytes.
 
-Thus the canonical address for new public key types would be `AddressHash(proto.MessageName(pk), pk.Bytes)`.
-
-### Multisig Addresses
+### Named Accounts
 
-For new multisig public keys, we define a custom address format not based on any encoding scheme
-(amino or protobuf). This avoids issues with non-determinism in the encoding scheme. It also
-ensures that multisig public keys which differ simply in the ordering of keys have the same
-address by sorting child public keys first.
-
-First we define a proto message for multisig public keys:
-```proto
-package cosmos.crypto.multisig;
-
-message PubKey {
-  uint32 threshold = 1;
-  repeated google.protobuf.Any public_keys = 2;
-}
-```
+Some accounts can have a fixed name. We were discussing an idea of an account with a predefined name (eg: `me.regen`), this could be used by institutions or modules.
+Without going into details, this kind of addresses are compatible with the hash based addresses discussed here as long as they don't conflict with the same length.
+More specifically, any special account, with custom naming, must not have length equal 20 byte nor 32 bytes.
 
-We define the following `Address()` function for this public key:
+### Module Addresses
 
-```
-func (multisig PubKey) Address() {
-  // first gather all the addresses of each nested public key
-  var addresses [][]byte
-  for key := range multisig.Keys {
-    addresses = append(joinedAddresses, key.Address())
-  }
-
-  // then sort them in ascending order
-  addresses = Sort(addresses)
-
-  // then concatenate them together
-  var joinedAddresses []byte
-  for addr := range addresses {
-    joinedAddresses := append(joinedAddresses, addr...)
-  }
-
-  // form the string prefix from the message name (cosmos.crypto.multisig.PubKey) and the threshold joined together
-  prefix := fmt.Sprintf("%s/%d", proto.MessageName(multisig), multisig.Threshold)
-
-  // use the standard AddressHash function
-  return AddressHash(prefix, joinedAddresses)
-}
-``` 
+TODO
```

Comment:
What is our general strategy for generating module accounts? Is this intended to be articulated in this PR, or will it come in a follow-up at a later date?

Reply:
We will need to update it. We are working on it: https://github.com/cosmos/cosmos-sdk/issues/8041#issuecomment-736627111

Reply:
I described it in the _Simple Account Address_ section